compose/cli/main.py

```diff
@@ -582,6 +574,21 @@ def version(self, project, options):
             print(get_version_info('full'))
 
 
+def convergence_strategy_from_opts(options):
+    allow_recreate = not options['--no-recreate']
+    force_recreate = options['--force-recreate']
+    if force_recreate and not allow_recreate:
+        raise UserError("--force-recreate and --no-recreate cannot be combined.")
+
+    if force_recreate:
+        return ConvergenceStrategy.always
+
+    if not allow_recreate:
+        return ConvergenceStrategy.never
+
+    return ConvergenceStrategy.changed
```

Comment:
It makes me think we should've gone with a `--strategy` option, rather than multiple boolean options, but that's a discussion for another time